packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -257,6 +261,16 @@ export default createRule<Options, MessageId>({
         return { isUnhandled: true, promiseArray: true };
       }
 
+      // await expression addresses promises, but not promise arrays.
+      if (node.type === AST_NODE_TYPES.AwaitExpression) {
+        // you would think this wouldn't be strictly necessary, since we're
+        // anyway checking the type of the expression, but, unfortunately TS
+        // reports the result of `await (promise as Promise<number> & number)`
```

Comment:
Typescript gives a warning
> 'await' has no effect on the type of this expression.

Lets not include example of those cases for which the ts compiler gives us a warning/error (?)

[Minimum repro for the warning](https://www.typescriptlang.org/play/?target=9#code/IYZwngdgxgBAZgV2gFwJYHsIwELAF54AUAlDAN4CwAUDLTADYCmy866MAvDMAO7CotCABQBO6ALaoQjAHQjGIdPQBujQgFZSoGKIlTGAHggJxAI0YiAfDABkMY2YsxiAbmp0YAek90AegH5qAF8gA) , hover over `await` keyword in line 2.

Reply:
> Lets not include example of those cases for which the ts compiler gives us a warning/error (?)

No, we'll still have to handle them correctly. Especially when it's just a warning like this one.